test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -759,6 +759,86 @@ TEST_F(InjaTransformerTest, UseBodyFunction) {
   EXPECT_EQ(body.toString(), "1 1");
 }
 
+TEST_F(InjaTransformerTest, MergeJsonKeys) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+  TransformationTemplate transformation;
+
+
+  // TransformationTemplate transformation;
+  // transformation.mutable_body()->set_text("{\"ext2\": \"{{header(\":path\")}\" }");
+
+  envoy::api::v2::filter::http::InjaTemplate inja_template;
+  inja_template.set_text("\"{{header(\":path\")}}\"");
+  envoy::api::v2::filter::http::MergeJsonKeys_OverridableTemplate tmpl;
+  (*tmpl.mutable_tmpl()) = inja_template;
+  (*transformation.mutable_merge_json_keys()->mutable_json_keys())["ext2"] = tmpl;
```

Comment:
please add tests for the dot logic. include edge cases to make sure we don't crash. i.e.
- "foo.bar"
- "foo..bar"
- ".foo."
- "..."
- ".foo
- "bar."

Reply:
also empty string?

Reply:
Discussed offline, I am going to remove this ability for now as it's not necessary. In order to ensure we can potentially add it back later I will throw an exception at `config_time` if a `"."` is present in the key